File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());
        int maxLength = 0;

        for (int num : nums) {
            // Check if it's the start of a sequence
            if (numSet.find(num - 1) == numSet.end()) {
                int currentNum = num;
                int currentStreak = 1;

                // Count the length of the sequence
                while (numSet.find(currentNum + 1) != numSet.end()) {
                    currentNum++;
                    currentStreak++;
                }

                // Update the maximum length
                maxLength = max(maxLength, currentStreak);
            }
        }

        return maxLength;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.js
var longestConsecutive = function(nums) {
    let set = new Set(nums);   // Add everything first

    let longest = 0;

    for (let num of set) {     // Iterate over set (avoid duplicates)

        // Start only if it is the beginning of a sequence
        if (!set.has(num - 1)) {
            let currentNum = num;
            let count = 1;

            while (set.has(currentNum + 1)) {
                currentNum++;
                count++;
            }

            longest = Math.max(longest, count);
        }
    }

    return longest;
};